app/scripts/controllers/app-state.js

```diff
@@ -122,6 +126,8 @@ export default class AppStateController extends EventEmitter {
       }
       this.emit(METAMASK_CONTROLLER_EVENTS.UPDATE_BADGE);
     }
+
+    this._acceptApprovalRequest();
```

Comment:
Perhaps worth a comment here confirming all approval requests are automatically rejected by the background if the unlock is ignored and the popup closed? I've assumed that is the current behaviour, worth verifying.

Reply:
yes, it rejects and updates the badge number when the popup is closed.